src/sliceList.ml

```diff
@@ -0,0 +1,195 @@
+(* Double-linked list for burrow slices which acts as a higher-level interface for data stored in the AVL queue.
+ * This data structure allows for fast lookups of slices for a specific burrow, and
+ * functions which are adding or removing slices from the queue should use this module instead of
+ * AVL.ml directly since this module will automatically ensure that the burrow slice lists stay up
+ * to date.
+*)
+open Avl
+open Mem
+open LiquidationAuctionTypes
+open LiquidationAuctionPrimitiveTypes
+
+type slice_list_element = SliceListElement of (leaf_ptr * liquidation_slice)
+[@@deriving show]
+
+let slice_list_element_contents (e: slice_list_element) =
+  match e with
+  | SliceListElement (_, contents) -> contents.contents
+
+type slice_list_bounds = {
+  slice_list_youngest_ptr : leaf_ptr;
+  slice_list_oldest_ptr : leaf_ptr;
+}
+[@@deriving show]
+
+type slice_list_meta = {
+  slice_list_burrow: burrow_id;
+  slice_list_bounds: slice_list_bounds option
+}
+[@@deriving show]
+
+(* Question: Is it worth storing one of the end elements within the SliceList?  *)
+type slice_list = SliceList of slice_list_meta
+[@@deriving show]
+
+let slice_list_empty (burrow: burrow_id) = SliceList {slice_list_burrow=burrow; slice_list_bounds=(None:slice_list_bounds option);}
+
+let slice_list_is_empty (l: slice_list) =
+  let meta = match l with SliceList meta -> meta in
+  match meta.slice_list_bounds with
+  | Some _ -> false
+  | None -> true
+
+(* Constructs a burrow slice list for the given burrow id using the provided auction state *)
+let slice_list_from_auction_state (auctions: liquidation_auctions) (burrow_id: burrow_id) =
```

Comment:
If some of these functions are used at most once per entrypoint, we can add a `[@inline]` pragma to them which can sometimes reduce the codebase.